Wykonano zadanie

a-dreszler opened the pull request:
(Zmieniłem w teście 100 na 1000 bo był błąd :))

a-dreszler committed:
Wykonano zadanie
(Zmieniłem w teście 100 na 1000 bo był błąd :))

marcinkunert commented on src/main/java/pl/javastart/task/CustomList.java:
```diff
@@ -1,6 +1,79 @@
+
+    private void expandArray() {
+        array = Arrays.copyOf(array, array.length + 1);
+    }
```
Zmiana rozmiaru tablicy za każdym razem jest mało efektywna. Lepiej np przy dodawaniu podwajać rozmar tablicy, jeśli brakuje miejsca.

marcinkunert commented on src/main/java/pl/javastart/task/CustomList.java:
```diff
@@ -1,6 +1,79 @@
+        for (int i = array.length - 1; i > index; i--) {
+            array[i] = array[i - 1];
+        }
+    }
```
Tutaj lepiej zastoswać System.arrayCopy, który skopiuje całą tablicę na raz.

a-dreszler committed:
Wprowadzono poprawki:
 > CustomList:
   > od teraz tablica jest inicjalizowana w konstruktorze tablicą o domyślnym rozmiarze INITIAL_ARRAY_SIZE
   > tablica rozszerzana jest tylko gdy ilość elementów w niej równa jest rozmiarowi, i to dwukrotnie aby nie wykonywać kopiowania całej tablicy przy każdym dodawaniu elementu
   > usunięto metodę trimArray aby kopiowanie tablicy nie zachodziło przy każdym wywołaniu funkcji remove() - względy wydajnościowe
   > metody shiftRight i shiftLeft od teraz wykorzystują System.arraycopy
   > warunek rzucenia IndexOutOfBoundsException sprawdzany teraz jest względem pola elementNumber a nie długości tablicy array
   > toString ma teraz własną implementację wyświetlającą tylko elementy tablicy od 0 do elementNumber - 1
